### PR TITLE
Add support for GraphQL via WPGraphQL hooks

### LINKED ISSUE #150: Add support for GraphQL via WPGraphQL hooks
### Description

I would incorporate support for WPGraphQL using it's custom hooks so WP Curate can be used with GraphQL 

### Use Case

When a user installs WP Curate it should provide support for the WPGraphQL plugin with helpful additions for creating custom GraphQL queries

### PR DESCRIPTION
**Pull request by ellm**

### CODE REVIEW

File: src/features/class-graphql.php
Comment (kevinfodness) [inline src/features/class-graphql.php]: Add type to function parameter also:
```suggestion
	private function get_graphql_type_by_post_type( string $post_type_string ): string {
```

File: src/features/class-graphql.php
Comment (kevinfodness) [inline src/features/class-graphql.php:37]: This check should come first, because it doesn't rely on fetching data from somewhere else (e.g., get_post_type_object).

File: src/features/class-graphql.php
Comment (kevinfodness) [inline src/features/class-graphql.php]: Remove `opinion` here, that's a custom post type that we can't expect to exist. The call to this filter should mirror the other ones in the plugin, which default to only `post`:
```suggestion
		$this->allowed_post_types = apply_filters( 'wp_curate_allowed_post_types', [ 'post' ] );
```

Comment (renatonascalves) [inline src/features/class-graphql.php]: The `opinion` post type looks client specific, no?

File: src/features/class-graphql.php
Comment (kevinfodness) [inline src/features/class-graphql.php]: The way the hook system works means that we don't actually need to see if the WPGraphQL class exists at all, we can register a callback for a graphql hook, which won't fire if the plugin isn't installed and activated:
```suggestion
		add_action( 'graphql_register_types', [ $this, 'graphql_register_types' ] );
```

**Review by kevinfodness — APPROVED** [review]
A few minor things to fix up before merge, otherwise 🍣

File: src/features/class-graphql.php
Comment (ellm) [reply src/features/class-graphql.php]: Yes! I left this by mistake after doing some testing locally. Thanks for flagging.

Comment (ellm) [reply src/features/class-graphql.php]: Thanks for flagging - this was left in by mistake after doing some local testing.

File: src/features/class-graphql.php
Comment (ellm) [reply src/features/class-graphql.php]: Ah yes, makes sense.

File: src/features/class-graphql.php
Comment (ellm) [reply src/features/class-graphql.php:37]: Updated in d6644c4e6727953425c9ad0899f4de0f041f1806

**Comment by ellm** [comment]
Thanks @kevinfodness ! Any chance you know how best to fix the phpstan notices? Looks like it is flagging the WPGraphQL functions b/c there are no references to them in the codebase

**Comment by kevinfodness** [comment]
@ellm The "right way" to fix those notices would be to install the WPGraphQL plugin as part of CI and add scandirs config to PHPStan to read in that content, but it would also require installing WPGraphQL as essentially a peer dependency when developing locally, and for what we're doing here, I'm not sure that's worth it. You have my blessing to `@phpstan-ignore` the parts that are complaining about not understanding the WPGraphQL functionality.

**Review by kevinfodness — APPROVED** [review]
🍣

**Comment by ellm** [comment]
Thanks @kevinfodness - I also added some initial documentation on this in the Wiki - https://github.com/alleyinteractive/wp-curate/wiki/WPGraphQL-Integration

**Pull request merged**
